File: Arrays&Hashing/FizzBuzz.py
# https://leetcode.com/problems/fizz-buzz/submissions/977227230/
def fizzBuzz(n: int) -> list[str]:
    answer = []
    for i in range(1,n+1):
        if i % 3 == 0 and i % 5 == 0:
            answer.append('FizzBuzz')
        elif i % 3 == 0:
            answer.append('Fizz')
        elif i % 5 == 0:
            answer.append('Buzz')
        else:
            answer.append(str(i))
    return answer

print(fizzBuzz(3)) # ["1","2","Fizz"]
print(fizzBuzz(5)) # ["1","2","Fizz","4","Buzz"]
print(fizzBuzz(15)) # ["1","2","Fizz","4","Buzz","Fizz","7","8","Fizz","Buzz","11","Fizz","13","14","FizzBuzz"]

File: Arrays&Hashing/FizzBuzz.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;


class FizzBuzz {
  public List<String> fizzBuzz(int n) {
      List<String> answer = new ArrayList<String>();

      for (int num = 1; num <= n; num++) {

      boolean divisibleBy5 = (num % 5 == 0);
      boolean divisibleBy3 = (num % 3 == 0);

          if (divisibleBy5 && divisibleBy3) {
              answer.add("FizzBuzz");
          } else if (divisibleBy3) {
              answer.add("Fizz");
          } else if (divisibleBy5) {
              answer.add("Buzz");
          } else {
              answer.add(Integer.toString(num));
          }
      }

      return answer;

    }
    public static void main(String[] args) {
      FizzBuzz fb = new FizzBuzz();
      var scanner = new Scanner(System.in);
      System.out.println("Enter a number to get FizzBuzzed:");
      var input = scanner.nextInt();
      scanner.close();
      System.out.println(fb.fizzBuzz(input));
    }
}